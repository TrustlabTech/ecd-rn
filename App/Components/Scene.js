import React, { Component } from 'react'
import { View } from 'react-native'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import * as appActions from '../Actions/App'
import WaitModal from '../Components/WaitModal'

class Scene extends Component {

  constructor(props) {
    super(props)
    this.actions = this.props.actions
    this.dispatch = this.props.dispatch
  }

  render() {
    const { modalWaiting, modalVisible, modalText } = this.props.state.App
    return (
      <View style={{flex: 1, backgroundColor: 'white'}}>
        <WaitModal
          animating={ modalWaiting }
          visible={ modalVisible }
          onPressClose={ () => this.props.dispatch(appActions.setModal({modalVisible:false})) }
          text={ modalText }
          ref="waitmodal"
        />
          <View style={{flex: 1}}>
            {this.props.children}
          </View>

      </View>
    )
  }
}

export default connect(
  (state) => ({
    state: state
  }),
  (dispatch) => ({
    actions: bindActionCreators(appActions, dispatch)
  })
)(Scene)