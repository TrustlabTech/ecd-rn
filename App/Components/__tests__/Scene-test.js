import 'react-native'
import React from 'react'
import Scene from '../Scene'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <Scene/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})