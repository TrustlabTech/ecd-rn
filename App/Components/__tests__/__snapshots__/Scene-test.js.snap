exports[`test renders correctly loaded 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fefefe",
      "flex": 1,
    }
  } />
`;

exports[`test renders correctly not loaded 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail">
  Page is loading
</Text>
`;
